samples/sensor/qdec/src/main.c

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2022 Valerio Setti <valerio.setti@gmail.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/sys/printk.h>
+
+void main(void)
+{
+	struct sensor_value val;
+	int rc;
+	const struct device *dev = DEVICE_DT_GET(DT_ALIAS(qdec0));
+
+	if (!device_is_ready(dev)) {
+		printk("Qdec device is not ready\n");
+		return;
+	}
+
+	printk("Quadrature decoder sensor test\n");
+
+	while (true) {
+		rc = sensor_sample_fetch(dev);
+		if (rc != 0) {
+			printk("Failed to fetch sample (%d)\n", rc);
+			return;
+		}
+
+		rc = sensor_channel_get(dev, SENSOR_CHAN_ROTATION, &val);
+		if (rc != 0) {
+			printk("Failed to get data (%d)\n", rc);
+			return;
+		}
+
+		printk("Value = %d", val.val1);
```

Comment:
```suggestion
		printk("Position = %d degrees", val.val1);
```

Reply:
At first I was going to write it the same way you proposed, but then I thought that so far not all the quadrature decoder drivers return an angle. For example I don't think the "drivers/sensor/qdec_sam does; it seems to me that it just returns the raw counter value. Therefore I was not sure of what was the best solution here...
Should I assume that from now on all the decoder driver will return an angle?

Reply:
@gmarull 
I did some other check on the other qdec drivers and I found that the behavior is not really homogeneous:
- "drivers/sensor/qdec_sam/" --> returns the raw value of the counter on val1 and nothing on val2
- "drivers/sensor/qdec_nrfx/" --> returns a differential value (compared to the previous read) with an integer number of turns on val1 and the remainder on val2
- mine, according to the required changes of this PR, returns the absolute angle position on val1 and nothing on val2

I don't know if the change you suggested applies well here... What do you think?

Reply:
Yup, that's indeed a problem that needs to be resolved. If a driver supports the rotation channel, it should provide value in degrees, not counts. Driver could support 2 channels, though, one for counts and another for degrees. About using val1/val2: it's all about the resolution driver supports. When using counts, val2 doesn't make sense, but when using degrees it may be used if you have a resolution < 1 degree.